File: app/src/main/java/com/zagori/bottomnavbar/MainActivity.java
package com.zagori.bottomnavbar;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

import android.view.MenuItem;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    private TextView mTextMessage;

    private BottomNavBar.OnBottomNavigationListener mOnBottomNavItemSelectedListener =
            new BottomNavBar.OnBottomNavigationListener() {
                @Override
                public boolean onNavigationItemSelected(MenuItem menuItem) {
                    switch (menuItem.getItemId()){
                        case R.id.navigation_home:
                            mTextMessage.setText(R.string.title_home);
                            return true;
                        case R.id.navigation_payment:
                            mTextMessage.setText(R.string.title_payment);
                            return true;
                        case R.id.navigation_new_cart:
                            mTextMessage.setText(R.string.title_new_cart);
                            return true;
                        case R.id.navigation_dashboard:
                            mTextMessage.setText(R.string.title_dashboard);
                            return true;
                        case R.id.navigation_notifications:
                            mTextMessage.setText(R.string.title_notifications);
                            return true;
                    }
                    return false;
                }
            };


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mTextMessage = findViewById(R.id.message);
        BottomNavBar bottomNavView = findViewById(R.id.bottom_nav_view);
        bottomNavView.setBottomNavigationListener(mOnBottomNavItemSelectedListener);
    }
}

File: sample/src/main/java/com/zagori/bottomnavbar/sample/MainActivity.kt
package com.zagori.bottomnavbar.sample

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.zagori.bottomnavbar.BottomNavBar.OnBottomNavigationListener
import com.zagori.bottomnavbar.sample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private val binding by lazy { ActivityMainBinding.inflate(layoutInflater) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)

        binding.bottomNavView.setBottomNavigationListener(object: OnBottomNavigationListener{
            override fun onNavigationItemSelected(menuItem: MenuItem?): Boolean {
                return when (menuItem!!.itemId) {
                    R.id.navigation_home -> {
                        binding.message.setText(R.string.title_home)
                        true
                    }
                    R.id.navigation_payment -> {
                        binding.message.setText(R.string.title_payment)
                        true
                    }
                    R.id.navigation_new_cart -> {
                        binding.message.setText(R.string.title_new_cart)
                        true
                    }
                    R.id.navigation_dashboard -> {
                        binding.message.setText(R.string.title_dashboard)
                        true
                    }
                    R.id.navigation_notifications -> {
                        binding.message.setText(R.string.title_notifications)
                        true
                    }
                    else -> false
                }
            }
        })
    }
}